blocks/library/block/edit-panel/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+function ReusableBlockEditPanel( props ) {
+	const { isEditing, name, isSaving, onEdit, onDetach, onChangeName, onSave, onCancel } = props;
+
+	return (
+		<div className="reusable-block-edit-panel">
+			{ ! isEditing && ! isSaving && [
+				<span key="info" className="reusable-block-edit-panel__info">
+					<b>{ name }</b>
+				</span>,
+				<Button
+					key="edit"
+					isLarge
+					className="reusable-block-edit-panel__button"
+					onClick={ onEdit }>
+					{ __( 'Edit' ) }
+				</Button>,
+				<Button
+					key="detach"
+					isLarge
+					className="reusable-block-edit-panel__button"
+					onClick={ onDetach }>
+					{ __( 'Detach' ) }
+				</Button>,
+			] }
+			{ ( isEditing || isSaving ) && [
```

Comment:
Should the editing variation be visible while the name is being saved? In my testing, it flips back to the static text immediately upon hitting Save.

Reply:
Good catch—that's a bug that must have slipped in during a merge. Fixed in a820d1cc43dddd97d44e6e756707dfc6ab56c80b.